src/Components/events/DetailEvent.js

```diff
@@ -20,26 +16,20 @@ const DetailEvent = () => {
   const eventId = useParams().id;
   const location = useLocation();
   const userId = location.state.userId;
+  const eventsList = location.state.eventsList;
   const [goToCreateNewEvent, setGoToCreateNewEvent] = useState(false);
   const [goToDashboard, setGoToDashboard] = useState(false);
   const [goToEditEvent, setGoToEditEvent] = useState(false);
   const [eventToEdit, setEventToEdit] = useState("");
-  const [eventsList, setEventsList] = [];
-
-  useEffect(() => {
-    async function getEvents() {
-      const eventsFound = await getEventsFromServer();
-      setEventsList(eventsFound.events);
-    }
-    getEvents();
-  });
+  const textButtonEvent = getTextButton(userId, eventId);
+  const [textButton, setTextButton] = useState(textButtonEvent);
 
   const drawEvent = () => {
-    if (!eventId || !eventsList) {
+    if (!eventId || !eventsList || eventsList === []) {
       return;
     } else {
       const event = eventsList.find((event) => event.id.toString() === eventId);
```

Comment:
remove this filter from here, we should already have done this when we call getEventFromServer